domain/src/main/java/org/oppia/domain/exploration/ExplorationDataController.kt

```diff
@@ -0,0 +1,289 @@
+package org.oppia.domain.exploration
+
+import android.content.Context
+import androidx.lifecycle.LiveData
+import org.json.JSONArray
+import org.json.JSONException
+import javax.inject.Inject
+import javax.inject.Singleton
+import org.oppia.app.model.Exploration
+import org.oppia.util.data.AsyncResult
+import java.io.IOException
+import org.json.JSONObject
+import org.oppia.app.model.AnswerGroup
+import org.oppia.app.model.Outcome
+import org.oppia.app.model.Interaction
+import org.oppia.app.model.InteractionObject
+import org.oppia.app.model.RuleSpec
+import org.oppia.app.model.State
+import org.oppia.app.model.StringList
+import org.oppia.app.model.SubtitledHtml
+import org.oppia.util.data.DataProviders
+
+const val TEST_EXPLORATION_ID_0 = "test_exp_id_0"
+const val TEST_EXPLORATION_ID_1 = "test_exp_id_1"
+private const val WELCOME_EXPLORATION_DATA_PROVIDER_ID = "WelcomeExplorationDataProvider"
+private const val ABBOUT_OPPIA_EXPLORATION_DATA_PROVIDER_ID = "AboutOppiaExplorationDataProvider"
+
+/** Controller for retrieving an exploration. */
+@Singleton
+class ExplorationDataController @Inject constructor(private val context: Context,
```

Comment:
Nit: prefer wrapping parameters on a newline per breaking at the higher syntactical element (the '(' in this case: https://google.github.io/styleguide/javaguide.html#s4.5.1-line-wrapping-where-to-break), e.g.:

```kotlin
class ExplorationDataController @Inject constructor(
  private val context: Context, private val dataProviders: DataProviders
) {
```

Reply:
Done